test/testdata/rewriter/flatten_nested_sclass.rb

```diff
@@ -9,7 +9,7 @@ def bar; end
 end
 
 A.foo
-A.singleton_class.bar
```

Comment:
Is this not a regression? We were wrong before?

Reply:
```ruby
irb(main):001:0> class A
irb(main):002:1>   def self.foo;
irb(main):003:2>     class << self
irb(main):004:3>       def bar; end
irb(main):005:3>     end
irb(main):006:2>   end
irb(main):007:1> end
=> :foo
irb(main):008:0> A.foo
=> :bar
irb(main):009:0> A.bar
=> nil
```